website_compassion/controllers/my_account.py

```diff
@@ -196,8 +198,54 @@ def my_child(self, child_id=None, **kwargs):
         else:
             return request.redirect(f"/my/children?child_id={children[0].id}")
 
+    @route("/my/information", type="http", auth="user", website=True)
+    def my_information(self, form_id=None, **kw):
+        partner = request.env.user.partner_id
+
+        # Load forms
+        form_success = False
+        kw["form_model_key"] = "cms.form.partner.my.coordinates"
+        coordinates_form = self.get_form("res.partner", partner.id, **kw)
+        if form_id is None or form_id == coordinates_form.form_id:
+            coordinates_form.form_process()
+            form_success = coordinates_form.form_success
+
+        kw["form_model_key"] = "cms.form.partner.delivery"
+        delivery_form = self.get_form("res.partner", partner.id, **kw)
+        if form_id is None or form_id == delivery_form.form_id:
+            delivery_form.form_process()
+            form_success = delivery_form.form_success
+
+        values = self._prepare_portal_layout_values()
+        values.update({
+            "partner": partner,
+            "coordinates_form": coordinates_form,
+            "delivery_form": delivery_form,
+        })
+
+        # This fixes an issue that forms fail after first submission
+        if form_success:
+            result = request.redirect("/my/information")
```

Comment:
Maybe this is no more needed, did you try without? I guess we should be able to work with forms without this if we understood well how they work.
Maybe you also removed some javascript that would be useful to handle various forms in the same page (in popups).

Reply:
No, this does not work properly if I remove this part of the code (a manual refresh of the page is required to see the changes).